src/main.cpp

```diff
@@ -152,16 +154,46 @@ int main(int argc, char *argv[]){
 
             std::cout << "\n⭐ CPU analysis ⭐" << std::endl;
             auto program = parse_to_cfg_transformer(prog);
+            reduce_variables(program.entryNodes);
             Stopwatch cpu_watch;
-            cpu_analysis(program);
+            auto cpu_nodes = cpu_analysis(program);
             cpu_watch.save_time<Microseconds>();
 
             std::cout << "\n⭐ GPU cuBLAS analysis ⭐" << std::endl;
+            program = parse_to_cfg_transformer(prog);
             time_func("Cublas creation: ", 
                 create_cublas);
             Stopwatch cuBLAS_watch;
-            gpu_analysis(program.nodes);
+            reduce_variables(program.entryNodes);
+            auto cublas_nodes = gpu_analysis(program.nodes);
             cuBLAS_watch.save_time<Microseconds>();
+
+            if(!state_equality(cpu_nodes, cublas_nodes)){
+                std::cout << "###### cpu != cublas #####" << std::endl;
+            }
+
+            init_gpu();
```

Comment:
init_gpu should probably be called before cublas, to ensure fairness even if we dont quite know if it has an impact.